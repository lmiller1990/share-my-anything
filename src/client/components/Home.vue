<template>
  <div class="main banner container">
		<div>
			Share my
		</div>
		<input v-model="query" placeholder="cat" id="query-input" @keyup.enter="enterHandler">
  </div>
</template>

<script>
	import axios from 'axios'
	import { getImagesEndpointRoute } from '../../shared/routes'

  export default {
    name: 'Home',

		data() {
			return {
				query: null
			}
		},

		methods: {
			enterHandler(e) {
				axios(getImagesEndpointRoute(this.query))
					.then(res => console.log(res))
			}
		}
	}
</script>

<style scoped>
.main.banner.container {
	border: 1px solid grey;
	display: flex;
	flex-direction: row;
	font-size: 5em;
	justify-content: space-around;
}
</style>
