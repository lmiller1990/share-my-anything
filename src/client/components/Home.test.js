import Home from './Home.vue'
import { shallow, mount } from 'avoriaz'
import Vue from 'vue'

describe('Home.vue', () => {
	it('enterHandler should be called', () => {
		const wrapper = shallow(Home)
		wrapper.setMethods({ enterHandler: jest.fn() })
		wrapper.find('#query-input')[0].trigger('keyup.enter')
		expect(wrapper.vm.enterHandler.mock.calls.length).toBe(1)
	})

	it('enterHandler should not be called', () => {
		const wrapper = shallow(Home)
		wrapper.setMethods({ enterHandler: jest.fn() })
		wrapper.find('#query-input')[0].trigger('keydown.up')
		expect(wrapper.vm.enterHandler.mock.calls.length).toBe(0)
	})

	it('matches the snapshot', () => {
		const wrapper = mount(Home)
		expect(wrapper.$el).toMatchSnapshot()
	})
})
