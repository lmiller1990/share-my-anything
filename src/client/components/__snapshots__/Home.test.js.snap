// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue matches the snapshot 1`] = `undefined`;
